### ISSUE TITLE
Error when updating `text` property of text object with Kivy backend

### ISSUE THREAD

**Issue by Frimkron** [Fiona/Myrmidon#46]
I get this error when updating the `text` property of a text object using Kivy, but only after a while i.e. not every time. It happens for the "complex_example" game too, where it happens immediately.
```
INFO:kivy:[Base        ] Leaving application in progress...
 Traceback (most recent call last):
WARNING:kivy:stderr: Traceback (most recent call last):
   File "main.py", line 40, in <module>
WARNING:kivy:stderr:   File "main.py", line 40, in <module>
     Window()
WARNING:kivy:stderr:     Window()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Entity.py", line 143, in __init__
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Entity.py", line 143, in __init__
     Game.run_game()
WARNING:kivy:stderr:     Game.run_game()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Game.py", line 221, in run_game
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Game.py", line 221, in run_game
     cls.engine['window'].open_window()
WARNING:kivy:stderr:     cls.engine['window'].open_window()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/window/kivy/engine.py", line 145, in open_window
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/window/kivy/engine.py", line 145, in open_window
     self.kivy_app.run()
WARNING:kivy:stderr:     self.kivy_app.run()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/app.py", line 825, in run
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/app.py", line 825, in run
     runTouchApp()
WARNING:kivy:stderr:     runTouchApp()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/base.py", line 484, in runTouchApp
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/base.py", line 484, in runTouchApp
     EventLoop.window.mainloop()
WARNING:kivy:stderr:     EventLoop.window.mainloop()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/core/window/window_pygame.py", line 378, in mainloop
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/core/window/window_pygame.py", line 378, in mainloop
     self._mainloop()
WARNING:kivy:stderr:     self._mainloop()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/core/window/window_pygame.py", line 282, in _mainloop
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/core/window/window_pygame.py", line 282, in _mainloop
     EventLoop.idle()
WARNING:kivy:stderr:     EventLoop.idle()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/base.py", line 324, in idle
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/base.py", line 324, in idle
     Clock.tick()
WARNING:kivy:stderr:     Clock.tick()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/clock.py", line 483, in tick
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/clock.py", line 483, in tick
     self._process_events()
WARNING:kivy:stderr:     self._process_events()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/clock.py", line 615, in _process_events
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/clock.py", line 615, in _process_events
     event.tick(self._last_tick, remove)
WARNING:kivy:stderr:     event.tick(self._last_tick, remove)
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/clock.py", line 374, in tick
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/clock.py", line 374, in tick
     ret = callback(self._dt)
WARNING:kivy:stderr:     ret = callback(self._dt)
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Game.py", line 255, in app_loop_callback
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Game.py", line 255, in app_loop_callback
     entity._iterate_generator()
WARNING:kivy:stderr:     entity._iterate_generator()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Entity.py", line 167, in _iterate_generator
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/Entity.py", line 167, in _iterate_generator
     return_val = next(self._state_generators[self._current_state])
WARNING:kivy:stderr:     return_val = next(self._state_generators[self._current_state])
   File "/home/mark/venvs/space/lib/python3.4/site-packages/game_common-0.0.1-py3.4.egg/game_common/gui.py", line 20, in execute
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/game_common-0.0.1-py3.4.egg/game_common/gui.py", line 20, in execute
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/gfx/kivy/engine.py", line 429, in text
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/gfx/kivy/engine.py", line 429, in text
     self.destroy_text_image()
WARNING:kivy:stderr:     self.destroy_text_image()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/gfx/kivy/engine.py", line 444, in destroy_text_image
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/gfx/kivy/engine.py", line 444, in destroy_text_image
     self.image.destroy()
WARNING:kivy:stderr:     self.image.destroy()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/gfx/kivy/engine.py", line 342, in destroy
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/myrmidon/engine/gfx/kivy/engine.py", line 342, in destroy
     self.image.remove_from_cache()
WARNING:kivy:stderr:     self.image.remove_from_cache()
   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/core/image/__init__.py", line 482, in remove_from_cache
WARNING:kivy:stderr:   File "/home/mark/venvs/space/lib/python3.4/site-packages/kivy/core/image/__init__.py", line 482, in remove_from_cache
     pat = type(f)(u'%s|%d|%d')
WARNING:kivy:stderr:     pat = type(f)(u'%s|%d|%d')
 TypeError: NoneType takes no arguments
WARNING:kivy:stderr: TypeError: NoneType takes no arguments
```

**Comment by Frimkron** [comment]
The following example recreates the issue:
``` python
from myrmidon import Game, Entity
from myrmidon.consts import *


class Window(Entity):
    
    def execute(self):
        self.font = Game.load_font("contra.ttf",size = 50)
        self.fps_text = Game.write_text(1024/2, 768/2, font = self.font, alignment = ALIGN_CENTRE)
        while True:
            self.fps_text.text = "FPS " + str(Game.current_fps)
            yield

Game.define_engine(*(['kivy'] * 4))
Game.screen_resolution = (1024, 768)
Game.full_screen = False
Window()
```

**Comment by Fiona** [comment]
what the hell

**Comment by Fiona** [comment]
fixed with my powers of doing nothing
(mark wasn't using the right version of kivy, he hadn't checked out the fixed branch of kivy-as)

**Comment by safehammad** [comment]
Ouch! Glad it was an easy fix. All the more reason to expedite getting the
"fixed kivy" changes merged into upstream?
